controllers/useraccount/console_usersettings_test.go

```diff
@@ -0,0 +1,144 @@
+package useraccount
+
+import (
+	"context"
+	"fmt"
+	. "github.com/codeready-toolchain/member-operator/test"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"testing"
+)
+
+func TestDeleteConsoleSettingObjects(t *testing.T) {
+	t.Run("Object found by name and deleted", func(t *testing.T) {
+		// given
+		ctx := context.Background()
+		cm := &corev1.ConfigMap{
+			TypeMeta: metav1.TypeMeta{Kind: "ConfigMap"},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "user-settings-johnsmith",
+				Namespace: UserSettingNS,
+			},
+		}
+		cl := fake.NewClientBuilder().WithObjects(cm).Build()
+
+		// when
+		err := deleteResource(ctx, cl, "johnsmith", cm)
+
+		// then
+		require.NoError(t, err)
+		// check that the configmap doesn't exist anymore
+		err = cl.Get(ctx, client.ObjectKey{Name: "user-settings-johnsmith", Namespace: UserSettingNS}, &corev1.ConfigMap{})
+		require.True(t, errors.IsNotFound(err))
+	})
+	t.Run("Object found by label and deletes successfully", func(t *testing.T) {
+		// given
+		cm := &corev1.ConfigMap{
+			TypeMeta: metav1.TypeMeta{Kind: "ConfigMap"},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "user-settings-name-no-match",
+				Namespace: UserSettingNS,
+				Labels: map[string]string{
+					ConsoleUserSettingsIdentifier: "true",
+					ConsoleUserSettingsUID:        "johnsmith",
+				},
+			},
+		}
+		// create a noise objects where the labels don't match
+		noiseObject := &corev1.ConfigMap{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "ConfigMap",
+				APIVersion: "v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "user-settings-name-no-match-noise",
+				Namespace: UserSettingNS,
+				Labels: map[string]string{
+					ConsoleUserSettingsIdentifier: "true",
+				},
+			},
+		}
+		cl := test.NewFakeClient(t, cm, noiseObject)
```

Comment:
Same here.

Reply:
I don't think this needs to be changed, since its adding the object to the fake client.